lib/utils.js

```diff
@@ -471,9 +471,46 @@ async function removeAllSessionWebSocketHandlers (server, sessionId) {
   }
 }
 
+/**
+ * Verify whether the given application is compatible to the
+ * platform where it is going to be installed and tested.
+ *
+ * @param {string} app - The actual path to the application bundle
+ * @param {boolean} isSimulator - Should be set to `true` if the test will be executed on Simulator
+ * @returns {?boolean} The function returns `null` if the application does not exist or there is no
+ * `CFBundleSupportedPlatforms` key in its Info.plist manifest.
+ * `true` is returned if the bundle architecture matches the device architecture.
+ * @throws {Error} If bundle architecture does not match the device architecture.
+ */
+async function verifyApplicationPlatform (app, isSimulator) {
+  log.debug('Verifying application platform');
+
+  const infoPlist = path.resolve(app, 'Info.plist');
+  if (!await fs.exists(infoPlist)) {
+    log.debug(`'${infoPlist}' does not exist`);
+    return null;
+  }
+
+  const {CFBundleSupportedPlatforms} = await plist.parsePlistFile(infoPlist);
+  log.debug(`CFBundleSupportedPlatforms: ${JSON.stringify(CFBundleSupportedPlatforms)}`);
+  if (!_.isArray(CFBundleSupportedPlatforms)) {
+    log.debug(`CFBundleSupportedPlatforms key does not exist in '${infoPlist}'`);
+    return null;
+  }
+
+  const isAppSupported = (isSimulator && CFBundleSupportedPlatforms.includes('iPhoneSimulator'))
```

Comment:
since we only use `isAppSupported` for the sake of one conditional, why not just put the condition into the conditional and axe the constant altogether?

Reply:
IMHO it improves readability, since the condition is quite long

Reply:
cool, fine w/ me